dealii/petsc_trilinos/CMakeLists.txt

```
@@ -0,0 +1,65 @@
+##
+#  CMake script for the petsc_trilinos tutorial program:
+##
+
+# Set the name of the project and target:
+set(TARGET "petsc_trilinos")
+
+# Declare all source files the target consists of. Here, this is only
+# the one step-X.cc file, but as you expand your project you may wish
+# to add other source files as well. If your project becomes much larger,
+# you may want to either replace the following statement by something like
+#  file(GLOB_RECURSE TARGET_SRC  "source/*.cc")
+#  file(GLOB_RECURSE TARGET_INC  "include/*.h")
+#  set(TARGET_SRC ${TARGET_SRC}  ${TARGET_INC})
+# or switch altogether to the large project CMakeLists.txt file discussed
+# in the "CMake in user projects" page accessible from the "User info"
+# page of the documentation.
+set(TARGET_SRC
+  ${TARGET}.cc
+  )
+
+# At least in some cases, deal.II needs to use the MPI wrapper:
+if (MPI_CXX_COMPILER)
+    set(CMAKE_CXX_COMPILER ${MPI_CXX_COMPILER})
+endif()
+
+# Usually, you will not need to modify anything beyond this point...
+
+cmake_minimum_required(VERSION 3.13.4)
+
+#
+# Are all dependencies fulfilled?
+#
+if(NOT ((DEAL_II_WITH_PETSC AND NOT DEAL_II_PETSC_WITH_COMPLEX) OR DEAL_II_WITH_TRILINOS) OR NOT DEAL_II_WITH_P4EST) # keep in one line
+  message(FATAL_ERROR "
+Error! This tutorial requires a deal.II library that was configured with the following options:
+    DEAL_II_WITH_PETSC = ON
+    DEAL_II_PETSC_WITH_COMPLEX = OFF
+    DEAL_II_WITH_P4EST = ON
+or
+    DEAL_II_WITH_TRILINOS = ON
+    DEAL_II_WITH_P4EST = ON
+However, the deal.II library found at ${DEAL_II_PATH} was configured with these options:
+    DEAL_II_WITH_PETSC = ${DEAL_II_WITH_PETSC}
+    DEAL_II_PETSC_WITH_COMPLEX = ${DEAL_II_PETSC_WITH_COMPLEX}
+    DEAL_II_WITH_P4EST = ${DEAL_II_WITH_P4EST}
+    DEAL_II_WITH_TRILINOS = ${DEAL_II_WITH_TRILINOS}
+This conflicts with the requirements.
+One or both of the aforementioned combinations of prerequisites are not met by your installation, but at least one is required for this tutorial step."
+    )
+endif()
+
+deal_ii_initialize_cached_variables()
+set(CLEAN_UP_FILES *.log *.gmv *.gnuplot *.gpl *.eps *.pov *.vtk *.ucd *.d2 *.vtu *.pvtu)
+add_executable(${TARGET} ${TARGET_SRC})
+deal_ii_setup_target(${TARGET})
+
+xsdk_add_test(
+    NAME
+    dealii-${TARGET}
+    MPI_NPROCS
+    1
+    COMMAND
+    $<TARGET_FILE:${TARGET}>
+)
```

See the comment in the suggestion.
```suggestion
if (NOT ENABLE_CUDA)
    # This example fails when using Spack-built `xsdk` with CUDA enabled, so it
    # is disabled for now.
    xsdk_add_test(
        NAME
        dealii-${TARGET}
        MPI_NPROCS
        1
        COMMAND
        $<TARGET_FILE:${TARGET}>
    )
endif()
```